<?xml version="1.0" ?>
<launch>
    <param name="/use_sim_time" value="true" />
    <group ns="cslibs_mapping_node_3d">
        <rosparam param="lasers_2d">
        </rosparam>
        <rosparam param="lasers_3d">
            ["/points"]
        </rosparam>

        <param name="subscriber_queue_size" value="93000" />
        <param name="base_frame" value="/odom" />
        <param name="rate" value="10.0" />
	<param name="output_path" value="" />

        <param name="occ_3d_grid_resolution" value="0.05" />
        <param name="occ_ndt_3d_grid_resolution" value="0.1" />
        <param name="ndt_3d_grid_resolution" value="0.1" />
	<param name="filter_laserscan3d" value="false" />
	<param name="filter_size" value="0.05" />

	<param name="ndt_2d_map_oru_active" value="false" />
	<param name="occ_2d_map_active" value="false" />
	<param name="occ_ndt_2d_map_active" value="false" />
	<param name="ndt_2d_map_active" value="false" />

	<param name="ndt_3d_map_oru_active" value="true" />
	<param name="occ_3d_map_active" value="false" />
	<param name="occ_ndt_3d_map_active" value="false" />
	<param name="ndt_3d_map_active" value="false" />

        <param name="ndt_oru_size_x" value="45.0" /><!-- geb: 45.0, fr: 300.0, new: 165.0-->
        <param name="ndt_oru_size_y" value="20.0" /><!-- geb: 20.0, fr: 170.0, new: 250.0-->
        <param name="ndt_oru_size_z" value="4.0" /> <!-- geb:  4.0, fr:  30.0, new:  60.0-->
        <param name="ndt_oru_cen_x" value="12.0" /> <!-- geb: 12.0, fr:  70.0, new:  82.5-->
        <param name="ndt_oru_cen_y" value="0.0" />  <!-- geb:  0.0, fr: -50.0, new: 100.0-->
        <param name="ndt_oru_cen_z" value="2.0" />  <!-- geb:  2.0, fr:   5.0, new:  20.0-->

	<param name="ndt_3d_map_oru_pub_active" value="true" />
	<param name="ndt_2d_map_oru_pub_active" value="true" />

        <param name="visibility_threshold" value="0.4" />
        <param name="prob_visible_if_occluded" value="0.2" />
        <param name="prob_visible_if_not_occluded" value="0.8" />
    </group>

    <node name="cslibs_mapping_node_3d" pkg="cslibs_mapping" type="cslibs_mapping_node_3d" output="screen" clear_params="true" />
</launch>
